src/users/services/sessions.service.ts

```diff
@@ -0,0 +1,37 @@
+import { Inject, forwardRef } from '@nestjs/common'
+import { User } from '@prisma/client'
+import { UsersService } from 'src/users/services/users.service'
+
+const SESSION_TIMEOUT = 1000 * 60 * 30 // 30 minutes (fullstory's inactivity timeout)
+
+export class SessionsService {
+  constructor(
+    @Inject(forwardRef(() => UsersService))
+    private readonly users: UsersService,
+  ) {}
+
+  async trackSession(user: User) {
+    const currentTime = Date.now()
+
+    // Get current metadata
+    const currentMetaData = user.metaData || {}
+    const lastVisited = currentMetaData.lastVisited || 0
+    const sessionCount = currentMetaData.sessionCount || 0
```

Comment:
can we add sessionCount to 
https://github.com/thegoodparty/gp-api/blob/develop/prisma/schema/user.jsonTypes.ts#L12